includes/data-stores/class-wc-data-store-wp.php

```diff
@@ -165,4 +165,41 @@ protected function exclude_internal_meta_keys( $meta ) {
 		return ! in_array( $meta->meta_key, $this->internal_meta_keys );
 	}
 
+	/**
+	 * Gets a list of props and meta keys that need updated based on change state
+	 * or if they are present in the database or not.
+	 *
+	 * @param  WC_Data $object              The WP_Data object (WC_Coupon for coupons, etc).
+	 * @param  array   $meta_key_to_props   A mapping of meta keys => prop names.
+	 * @param  string  $meta_type           The internal WP meta type (post, user, etc).
+	 * @return array                        A mapping of meta keys => prop names, filtered by ones that should be updated.
+	 */
+	protected function get_props_to_update( $object, $meta_key_to_props, $meta_type = 'post' ) {
+		$props_to_update = array();
+		$changed_props   = $object->get_changes();
+
+		// Normalize billing / shipping / other keys that might be part of a sub array.
```

Comment:
I've actually moved this up to WC_Order. It needs to act on get_changes.

https://github.com/woocommerce/woocommerce/pull/12950/files#diff-d40ff36ba2766206bbf9818c26167af8R344